Src/library/net/Clib/network.c

```diff
@@ -1188,74 +1192,75 @@ EIF_INTEGER c_get_sock_recv_timeout(EIF_INTEGER fd, EIF_INTEGER level)
 #ifdef EIF_WINDOWS
 	int arg;
 	asize = sizeof(arg);
-	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, SO_RCVTIMEO, (char *) &arg, &asize));
-	return (EIF_INTEGER) (arg / 1000); /* Convert milliseconds to seconds */
+	eif_net_check(getsockopt((EIF_SOCKET_TYPE) fd, (int) level, (int) optname, (char *) &arg, &asize));
+	return RTU64C(1000000) * arg; /* 1 millisecond = 1000000 nanoseconds */
```

Comment:
Can you add constants instead for all the variation of 10^x instead of having to put comments each time we use the value?